nncf/quantization/algorithms/tune_hyperparams/params_transformation.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import functools
+import itertools
+from copy import deepcopy
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+SearchSpace = Dict[str, Union[List[Any], "SearchSpace"]]
+
+
+def is_dataclass_instance(obj: Any):
+    return dataclasses.is_dataclass(obj) and not isinstance(obj, type)
+
+
+class ParamsTransformation:
```

Comment:
Do we really need such a complicated class just to create combinations of the parameters? Also, do we have examples of the resulting combinations?

Reply:
It was removed.